src/niri.rs

```diff
@@ -1120,7 +1121,21 @@ impl State {
                 self.niri.ipc_outputs_changed = true;
                 resized_outputs.push(output.clone());
             }
+
+            let background_color = config
+                .map(|c| c.background_color)
+                .unwrap_or(Color::from_array_unpremul(CLEAR_COLOR));
+
+            if self.niri.output_state.get(output).is_some_and(|os| {
+                os.background_buffer.color() != background_color.to_array_unpremul()
+            }) {
+                recolored_outputs.push((output.clone(), background_color));
+            }
```

Comment:
Can't this just set the color directly here instead of doing the whole `recolored_outputs`? Also avoid the `output_recolored()` function.

Reply:
I'll remove the `output_recolored()` function but I have used recolored_outputs because I cannot call `self.niri.queue_redraw(&output)` (mutable borrow) inside output loop which also borrows `self.niri` immutably. Is there a particular way of doing this? Thanks

Reply:
I guess you could leave `recolored_outputs` as a `Vec<Output>` for this to call `queue_redraw` after the loop.

Reply:
Pushed! Please let me know if everything is done correctly.